### PR TITLE
Add release notes and references for release-0.5

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Prepare branch for release.

#### Which issue(s) this PR fixes:

Part of #1256

#### Special notes for your reviewer:

This was based out of #958

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -229,7 +229,7 @@ ifndef ignore-not-found
   ignore-not-found = false
 endif
 
-clean-manifests = (cd config/components/manager && $(KUSTOMIZE) edit set image controller=gcr.io/k8s-staging-kueue/kueue:main)
+clean-manifests = (cd config/components/manager && $(KUSTOMIZE) edit set image controller=gcr.io/k8s-staging-kueue/kueue:release-0.4)
```
Comment (tenzen-y) [inline Makefile:232]:
```suggestion
clean-manifests = (cd config/components/manager && $(KUSTOMIZE) edit set image controller=gcr.io/k8s-staging-kueue/kueue:release-0.5)
```
Is this correct?

Comment (alculquicondor) [reply Makefile:232]: no :)

Comment (alculquicondor) [reply Makefile]: fixed

**Review by tenzen-y — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
/hold
I want do do a test

**Pull request merged**